lib/server/header/haproxy.js

```diff
@@ -0,0 +1,27 @@
+function parseHeader(header) {
+    var parts = header.split(' ')[2];
+    if (parts[0] !== 'PROXY') {
+        throw new Error('PROXY line missing, is haproxy is present?');
+    }
+    return parts[2];
+}
+
+function createParser(socket, connected, headerError) {
+    var buffer,
+        char;
```

Comment:
The `char` variable is not shared between executions, it should be in the function.